### PR TITLE
Better exception when not connected to a participant

### PR DESCRIPTION
**Pull request by viliam-durina**
Addresses this exception from #19319 and #19428 (the last one):
```
java.lang.NullPointerException
	at com.hazelcast.jet.impl.execution.SenderTasklet.call(SenderTasklet.java:114)
```

### CODE REVIEW

**Review by ufukyilmaz — APPROVED** [review]
It looks good to me

File: hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java [hunk 1]
```diff
@@ -43,23 +45,23 @@
...
  *             GenericRecord modifiedGenericRecord = genericRecord.cloneWithBuilder()
  *                     .setInt("age",22).build();
  *             entry.setValue(modifiedGenericRecord);
  *             return null;
  *         });
- * </pre>
+ * }</pre>
  * <p>
- * GenericRecord also allows to read from a cluster without having the classes on the client side.
+ * GenericRecord also allows reading from the cluster without having the classes on the client side.
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java:60]:
```suggestion
 * GenericRecord also allows reading from a cluster without having the classes on the client side.
```

I think "a" is more suitable here since we are talking about any cluster.

File: hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java [hunk 2]
```diff
@@ -43,23 +45,23 @@
...
  * For {@link Portable}, when {@link PortableFactory} is not provided in the config at the start,
  * a {@link HazelcastSerializationException} was thrown stating that a factory could not be found. Starting from 4.1,
- * the objects will be returned as {@link GenericRecord}. This way, the clients can read and write the objects back to
- * the cluster without needing the classes of the domain objects on the classpath.
+ * the objects will be returned as {@link GenericRecord}. This way, the clients can read and write objects back to
+ * the cluster without the need that the domain classes are on the classpath.
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java:64]:
```suggestion
 * the cluster without the need of having the domain classes on the classpath.
```

or 

```suggestion
 * the cluster without needing the domain classes on the classpath.
```

I prefer the latter one since it's simpler

**Review by srknzl — APPROVED** [review]
looks good from client's perspective.

left some suggestions about wording

File: hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java [hunk 3]
```diff
@@ -175,7 +176,11 @@ public void initialize(NodeEngineImpl nodeEngine,
         this.ptionArrgmt = new PartitionArrangement(partitionAssignment, nodeEngine.getThisAddress());
         Set<Integer> higherPriorityVertices = VertexDef.getHigherPriorityVertices(vertices);
         for (Address destAddr : remoteMembers.get()) {
-            memberConnections.put(destAddr, getMemberConnection(nodeEngine, destAddr));
+            Connection conn = getMemberConnection(nodeEngine, destAddr);
+            if (conn == null) {
+                throw new TopologyChangedException("no connection to job participant: " + destAddr);
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java:181]: is this exception propagated back to the client?

Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java:181]: This staff is called on [InitExecutionOperation](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/InitExecutionOperation.java#L92-L99). since it is a jet [AsyncOperation](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java#L50-L61) , it sends these exceptions as a response back to its caller (the job coordinator in this case).

Comment (frant-hartm) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java:181]: Is this hard to reproduce? A test would be great.

File: hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java [hunk 4]
```diff
@@ -28,13 +30,13 @@
 import java.util.Set;
 
 /**
- * A generic object interface that is returned to user when the domain class can not be created from any of the distributed
- * hazelcast data structures like {@link com.hazelcast.map.IMap} ,{@link com.hazelcast.collection.IQueue} etc.
+ * A generic object interface that is returned to the user when the domain class can not be created from any of the distributed
```
Comment (frant-hartm) [inline hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java:33]: I don't see how the javadoc changes and the missing connection to participant relate. Should this be split into two separate PRs?

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java (hunk 4 above)
Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java:33]: Yes, it should, but I was lazy. I'm fixing grammar as I read it, and then submit all the collected changes in the next _real_ PR i'm doing.

File: hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java (hunk 3 above)
Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java:181]: It's quite hard. The connection to the other member has to be already down, but a moments before when we're building the list of participants it still has to be there. There's a very small window, we could test it only with a test hook that will allow us to wait a while at a specific place.

**Comment by frant-hartm** [comment]
I have split the javadoc changes to https://github.com/hazelcast/hazelcast/pull/19495

**Review by frant-hartm — APPROVED** [review]
